### DISCUSSION TITLE
Night Vision FPS

### DISCUSSION THREAD

**Discussion by foureight84** [mrlt8/docker-wyze-bridge#609] [Q&A]
Is there a way to force Night Vision to stay at 30FPS?

I've noticed that when Night Vision is turned on, FORCE_FPS doesn't work on the camera and I can see in the logs that the FPS dropped:

```bash
2022/11/12 04:24:53 [Front Porch] WARNING: FPS param mismatch (avRecv FPS=15)
2022/11/12 04:24:53 [Front Porch] Requesting frame_rate=15
```

I know that with the [WZ Mini Hacks](https://github.com/gtxaspec/wz_mini_hacks/discussions/81), there is a flag to stop  FPS drop in night vision mode. But I assume they're able to do this by patching the iCamera process?

**Comment by mrlt8** [comment]
Does `FPS_FIX` work? Will have to look into this.

**Reply by foureight84** [reply]
It didn't. I forgot that I had that on as well. I just removed night vision for now. One observation I saw is that after turning Night Vision off, the camera timing started to drift. I saw it by as much as 30 seconds behind what I am seeing in the feed vs the clock on my phone.

**Reply by mrlt8** [reply]
Does the 30FPS function correctly during the day?

The drift is probably being caused by the mismatched FPS. It's a little tricky because the function returning the video data doesn't seem to display the correct FPS, so we may have to calculate the FPS manually and send the command to change it if it doesn't match what we're expecting.

**Reply by foureight84** [reply]
It does. It seems that the drift only happens for me on the Wyze Cam V3 after it enters and leaves night mode. What you mentioned seems exactly it from observation. Leaving night mode doesn't seem to change the FPS. There's also a large gap in the upper and lower end of the threshold to trigger Night Mode. There's a certain level of luminosity where the camera requires manual intervention. Even after manually enabling Night Vision and changing the settings in the Wyze app back to Auto, it stays in Night Vision when in fact it's not "dark enough" to automatically trigger.

**Reply by foureight84** [reply]
A quick update on the drifting issue. I can confirm with certainty now that it has nothing to do with the FPS drop due to Night Vision turning on. I am using frigate and it looks like the hardware acceleration flags I was using were not correct and caused unnecessary load on the CPU and the lag is a result of this. But the FPS drop to 15 still remains when Night Mode is running with `FPS_FIX` and `FORCE_FPS_CAM_NAME` set for ENV variables.

**Reply by foureight84** [reply]
I am currently using wyze-bridge in combination with wz-mini-hacks loaded on the device to enforce 30fps with night vision enabled and it seems to be working nicely. I was using wz-mini-hacks alone but I kept getting frame and audio drops from the RTSP with the client running in the camera. Everything seems a lot more stable with wz-bridge's stream copy.

**Reply by mrlt8** [reply]
Awesome! Sorry about the delay, just trying to get the current build stable, but I will look into adding the ability to force the FPS when night vision is active.

**Reply by foureight84** [reply]
> Awesome! Sorry about the delay, just trying to get the current build stable, but I will look into adding the ability to force the FPS when night vision is active.

No worries! Curious, are you able to output logs from the firmware to show stream connection issues? I'm using this in conjunction with Frigate and I notice that many times a day, the RTSP stream would disconnect. According to wyze-bridge logs, the wifi signal for all of my cameras are above 90%. I'm curious if this is a firmware issue or something caused/introduced by the wyze_mini_hacks.

**Reply by mrlt8** [reply]
I haven't really looked, but I believe wyze is able to pull the logs from the cameras, so we may be to pull them if they are doing so over the tutk protocol.
